Sorting/ShellSort.cpp

```diff
@@ -0,0 +1,62 @@
+/*
+    Shell sort
+    -----------
+    An in-place comparison sort which starts by sorting pairs of elements far apart from each other,
+    then progressively reducing the gap between elements to be compared. The time complexity depends on
+    the gap sequence, which is floor(N/2^K) in this implementation
+
+    Time complexity
+    ----------------
+    O(N^2), where N is the nubmer of elements.
+
+    Space complexity
+    -----------------
+    O(1).
+*/
+
+#include <iostream>
+#include <vector>
+#include "SortingUtils.hpp"
+
+using namespace std;
+
+void shell_sort (vector<int> &values, const bool &show_state) {
+    int n = values.size();
+
+    // We calculate the gap sequence in order of |n/2|, |n/4|,....
+    for (int gap = n/2; gap > 0; gap /= 2) {
+        for (int i = gap; i < n; ++i) {
+
+            int temp = values[i];
+            int j;
```

Comment:
Declare the variables `temp`, `i` and `j` outside the loops to prevent redeclaration on every iteration.